homeassistant/components/foscam/camera.py

```diff
@@ -111,6 +207,31 @@ def disable_motion_detection(self):
             _LOGGER.debug("Communication problem")
             self._motion_status = False
 
+    async def async_perform_ptz(self, movement, travel_time):
+        """Perform a PTZ action on the camera."""
+        _LOGGER.debug("PTZ action '%s' on %s", movement, self._name)
+
+        ret, _ = {
+            DIR_UP: self._foscam_session.ptz_move_up,
+            DIR_DOWN: self._foscam_session.ptz_move_down,
+            DIR_LEFT: self._foscam_session.ptz_move_left,
+            DIR_RIGHT: self._foscam_session.ptz_move_right,
+            DIR_TOPLEFT: self._foscam_session.ptz_move_top_left,
+            DIR_TOPRIGHT: self._foscam_session.ptz_move_top_right,
+            DIR_BOTTOMLEFT: self._foscam_session.ptz_move_bottom_left,
+            DIR_BOTTOMRIGHT: self._foscam_session.ptz_move_bottom_right,
+        }[movement]()
```

Comment:
We're in async context here. The library doesn't support asyncio. We have to schedule sync I/O on the executor thread pool. Use `self.hass.async_add_executor_job`.